spec/src/features/tree-spec.coffee

```diff
@@ -282,6 +282,179 @@ describe '"tree" control', ->
         $(this).click()
       expect($('.jstree-clicked').size()).toEqual(2)
 
+  describe "simplify_output option", ->
+    model = """
+      <prov:treeReference type="tree">
+        <prov:data_layer></prov:data_layer>
+      </prov:treeReference>
+    """
+    ui = """
+          <tree id="tree_to_simplify" ref="prov:treeReference" valueElementName="data_layer" cascade="true">
+            <item value="top_node_1">
+              <item value="level_1_child_1">
+                <item value="level_2_child_1">
+                  <item value="level_3_child_1"></item>
+                  <item value="level_3_child_2"></item>
+                </item>
+                <item value="level_2_child_2">
+                  <item value="level_3_child_3"></item>
+                  <item value="level_3_child_4"></item>
+                </item>
+              </item>
+            </item>
+            <item value="top_node_2" />
+              <item value="level_1_child_2">
+                <item value="level_2_child_2">
+                  <item value="relevant_child" relevant="true"></item>
+                  <item value="irrelevant_child" relevant="false"></item>
+                </item>
+                <item value="level_2_child_2">
+                  <item value="required_child" required="true"></item>
+                  <item value="not_required_child" required="false"></item>
+                </item>
+            </item>
+          </tree>
+        """
+    it "includes all selected and implied nodes when set to false", ->
+      no_simplify_ui = ui.replace('cascade="true"', 'cascade="true" simplify_output="false"')
+      form = template.form(dom, ui: no_simplify_ui, model: model, attrs: attrs)
+      $(':jstree').jstree('open_all')
+
+      $(":jstree li[node_value='level_2_child_1'] > a ").click()
+      expect(form.echoforms('serialize')).not.toMatch(/>top_node_1</)
+      expect(form.echoforms('serialize')).not.toMatch(/>level_1_child_1</)
+      expect(form.echoforms('serialize')).toMatch(/>level_2_child_1</)
+      expect(form.echoforms('serialize')).toMatch(/>level_3_child_1</)
+      expect(form.echoforms('serialize')).toMatch(/>level_3_child_2</)
+      expect(form.echoforms('serialize')).not.toMatch(/>level_2_child_2</)
+
+    it "defaults to true", ->
+      form = template.form(dom, ui: ui, model: model)
```

Comment:
I assume these three lines should be (and similar elsewhere in this file) should be `beforeEach` blocks.